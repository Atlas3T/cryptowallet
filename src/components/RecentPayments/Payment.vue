<template>
  <div class="columns is-mobile no-margin-bottom">
    <div class="column is-2">
      <div class="circle"/>
    </div>
    <div class="column">
      <div class="columns is-mobile no-margin-bottom">
        <div class="column is-7 no-padding-bottom">{{ payment.title }}</div>
        <div class="column no-padding-bottom">
          <Amount
            :amount="payment.amount"
            variant="['grey']"
          />
        </div>
      </div>
      <div class="grey-text">
        {{ paymentTime }}
      </div>
    </div>
  </div>
</template>

<script>
import Amount from './Amount.vue';

export default {
  name: 'Payment',
  components: {
    Amount,
  },
  props: {
    payment: {
      type: Object,
      required: true,
    },
  },
  computed: {
    /*
     * converts timestamp into full date string and then extracts only the time out of it (HH:mm)
     * @return {String} eg. "02:59"
     */
    paymentTime() {
      return new Date(this.payment.ts).toTimeString().split(' ')[0].slice(0, 5);
    },
  },
};
</script>

<style scoped>
.no-padding-bottom {
  padding-bottom: 0;
}

.no-margin-bottom {
  margin-bottom: 0;
}

.circle {
  background: #ececec;
  border-radius: 100%;
  height: 2.5rem;
  width: 2.5rem;
  margin-top: 5px;
}
</style>
