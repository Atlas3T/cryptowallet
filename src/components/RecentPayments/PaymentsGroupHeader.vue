<template>
  <div class="columns is-mobile payments-group-header">
    <div class="column">
      {{ group.date }}
    </div>
    <div class="column">
      <Amount :amount="balanceChange"/>
    </div>
  </div>
</template>

<script>
import Amount from './Amount.vue';

export default {
  name: 'PaymentsGroupHeader',
  components: {
    Amount,
  },
  props: {
    group: {
      type: Object,
      required: true,
    },
  },
  computed: {
    balanceChange() {
      return this.group.payments.reduce((accumulator, payment) => accumulator + payment.amount, 0);
    },
  },
};
</script>

<style scoped>
.payments-group-header {
  margin-bottom: -1rem;
  margin-top: 0.5rem;
}
</style>
