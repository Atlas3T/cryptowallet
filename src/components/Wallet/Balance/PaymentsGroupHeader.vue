<template>
  <div class="columns is-mobile payments-group-header">
    <div class="column">
      {{ group.date }}
    </div>
    <div class="column">
      <Amount :amount="balanceChange"/>
    </div>
  </div>
</template>

<script>
import Amount from './Amount.vue';

export default {
  name: 'PaymentsGroupHeader',
  components: {
    Amount,
  },
  props: {
    group: {
      type: Object,
      required: true,
    },
  },
  computed: {
    /*
     * adds up all of the paymements amounts within this payment group
     * will result in eiter a positive or a negative number
     * @return {Number}
     */
    balanceChange() {
      return this.group.payments.reduce((accumulator, payment) => { return accumulator + payment.amount; }, 0);
    },
  },
};
</script>

<style scoped>
.payments-group-header {
  margin-bottom: -1rem;
  margin-top: 0.5rem;
}
</style>
