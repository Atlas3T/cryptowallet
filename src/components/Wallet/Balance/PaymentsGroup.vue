<template>
  <section>
    <PaymentsGroupHeader :group="group" />
    <Payment
      v-for="payment in group.payments"
      :key="payment.ts"
      :payment="payment"
    />
  </section>
</template>

<script>
import PaymentsGroupHeader from './PaymentsGroupHeader.vue';
import Payment from './Payment.vue';

export default {
  name: 'PaymentsGroup',
  components: {
    PaymentsGroupHeader,
    Payment,
  },
  props: {
    group: {
      type: Object,
      required: true,
    },
  },
};
</script>
