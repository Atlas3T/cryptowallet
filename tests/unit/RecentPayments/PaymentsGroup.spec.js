import { shallowMount } from '@vue/test-utils';
import PaymentsGroup from '@/components/RecentPayments/PaymentsGroup.vue';
import Payment from '@/components/RecentPayments/Payment.vue';
import PaymentsGroupHeader from '@/components/RecentPayments/PaymentsGroupHeader.vue';

describe('PaymentsGroup.vue', () => {
  let wrapper;

  const propsData = {
    group: {
      date: '26 November',
      payments: [
        {
          ts: 1539527543000,
          title: 'Amazon',
          amount: -10.00,
        },
        {
          ts: 1539144743000,
          title: 'Motion',
          amount: -47.97,
        },
      ],
    },
  };

  function wrapperInit (options) {
    return shallowMount(PaymentsGroup, options);
  }

  function storeInit (custom) {
    wrapper = wrapperInit({ propsData });
  }

  beforeEach(() => storeInit());

  it('renders and matches snapshot', () => {
    const wrapper = wrapperInit({ propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a PaymentsGroupHeader component within it', () => {
    expect(wrapper.contains(PaymentsGroupHeader)).toBe(true);
  });

  it('loops through the data and renders enough of Payment components', () => {
    expect(wrapper.findAll(Payment).length).toBe(propsData.group.payments.length);
  });
});
