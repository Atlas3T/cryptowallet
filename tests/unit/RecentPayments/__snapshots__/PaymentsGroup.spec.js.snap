// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentsGroup.vue renders and matches snapshot 1`] = `
<section>
  <paymentsgroupheader-stub
    group="[object Object]"
  />
   
  <payment-stub
    payment="[object Object]"
  />
  <payment-stub
    payment="[object Object]"
  />
</section>
`;
